packages/core/src/new-api/types/artifact.ts

```diff
@@ -20,4 +20,5 @@ export interface Artifact {
  */
 export interface ArtifactResolver {
   load(contractName: string): Promise<Artifact>;
+  resolvePath(contractName: string): Promise<string>;
```

Comment:
Ideally we read the values instead of paths, as that offers more flexibility to implementers. 

I know this will change in another PR though.

Reply:
I have read the comments on this in the later PR. I think we need to walk through the memory and file implementations and then figure our how abstract vs how much delayed file resolution (reading the artifact on demand) we want.

I am going to close this comment in favour of that later discussion.